src/main/lrsql/spec/admin.clj

```diff
@@ -57,17 +62,42 @@
                    ::username
                    :lrsql.spec.admin.input/oidc-issuer]))
 
+(def query-admin-input-spec
+  (s/keys :req-un [::username
+                   ::password]))
+
+(def query-admin-by-id-input-spec
+  (s/keys :req-un [::account-id
+                   ::password]))
+
 (def query-validate-admin-input-spec
   (s/keys :req-un [::username
                    ::password]))
 
+(def query-validate-admin-by-id-input-spec
+  (s/keys :req-un [::account-id
+                   ::password]))
+
 (def admin-id-input-spec
   (s/keys :req-un [::account-id]))
 
 (def delete-admin-input-spec
   (s/merge admin-id-input-spec
            (s/keys :req-un [:lrsql.spec.admin.input/oidc-enabled?])))
 
+(def update-admin-password-params-spec
+  (s/and
+   (s/keys :req-un [::old-password
+                    ::new-password])
+   (fn new-pass-noteq-old-pass
+     [{:keys [old-password
+              new-password]}]
+     (not= old-password new-password))))
```

Comment:
Probably move this up to be with `admin-params-spec` and `admin-delete-params-spec` to avoid confusion between REST param specs and SQL query input specs.

Reply:
I moved it [SQL-187 move params spec to be with its siblings](https://github.com/yetanalytics/lrsql/pull/295/commits/5aaaea5eb702f40b9dcb56ac8f226c5997c3345d)

As for the other bit I think it is fine as part of the spec, but I'll add language to the form as you describe.

Reply:
See https://github.com/yetanalytics/lrs-admin-ui/pull/43/commits/a993ea36f874b58f26c042b656d60f03f7dc07e4